src/main/java/org/cardanofoundation/cip8/CIP8Parser.java

```diff
@@ -28,25 +28,65 @@
 import static org.cardanofoundation.ext.cbor.MoreCbor.serialize;
 import static org.cardanofoundation.ext.cose.COSEKey.deserialize;
 
+/**
+ * The {@code CIP8Parser} class is used to verify and parse a message that uses the CIP 8
+ * specification of the Cardano protocol.
+ * For further information on CIP 8, see <A href="https://cips.cardano.org/cips/cip8/">
+ * https://cips.cardano.org/cips/cip8/</A>.
+ */
 public final class CIP8Parser {
 
+    /**
+     * Static instance of {@code Logger} used for logging.
+     */
     private static final Logger logger = LoggerFactory.getLogger(CIP8Parser.class);
 
+    /**
+     * TODO
+     */
     private static final EdDSAParameterSpec ED_DSA_PARAMETER_SPEC = EdDSANamedCurveTable.getByName(ED_25519);
+
+    /**
+     * TODO
+     */
```

Comment:
We can mention that this is the the index in byte array see line 159 which corresponds to the beginning of public key